### PR TITLE
filmic: BUGFIX

### PR DESCRIPTION
**Pull request by aurelienpierre**
The SSE2 function getting the horizontal RGB max over the pixel failed and the output was inconsistent with the OpenCL version.

This is probably because the 4th channel is alpha and its value is random, so it could be identified as the max RGB.

We fall back to a pure C querying of the maximum, which is more reliable for now and almost as fast.

Also, take this opportunity to improve the tooltips text and make it more specific.

### CODE REVIEW

File: src/iop/filmic.c [hunk 1]
```diff
@@ -1777,7 +1775,8 @@ void gui_init(dt_iop_module_t *self)
   dt_bauhaus_slider_enable_soft_boundaries(g->global_saturation, 0.0, 1000.0);
   dt_bauhaus_slider_set_format(g->global_saturation, "%.2f %%");
   gtk_box_pack_start(GTK_BOX(self->widget), g->global_saturation, TRUE, TRUE, 0);
-  gtk_widget_set_tooltip_text(g->global_saturation, _("desaturates the output"));
+  gtk_widget_set_tooltip_text(g->global_saturation, _("desaturates the input of the module globally.\n"
+                                                      "you need to set this value below 100 \% \nif the chrominance preservation is enabled."));
```
Comment (TurboGit) [inline src/iop/filmic.c]: should be %%

File: src/iop/filmic.c [hunk 2]
```diff
@@ -1693,7 +1683,7 @@ void gui_init(dt_iop_module_t *self)
   gtk_box_pack_start(GTK_BOX(self->widget), g->grey_point_source, TRUE, TRUE, 0);
   dt_bauhaus_slider_set_format(g->grey_point_source, "%.2f %%");
   gtk_widget_set_tooltip_text(g->grey_point_source, _("adjust to match the average luminance of the subject.\n"
-                                                      "except in back-lighting situations, this should be around 18%."));
+                                                      "except in back-lighting situations, this should be around 18 \%."));
```
Comment (TurboGit) [inline src/iop/filmic.c]: should be 18%%

File: src/iop/filmic.c (hunk 1 above)
Comment (aurelienpierre) [reply src/iop/filmic.c]: done

File: src/iop/filmic.c (hunk 2 above)
Comment (aurelienpierre) [reply src/iop/filmic.c]: done

File: src/iop/filmic.c [hunk 3]
```diff
@@ -1693,7 +1683,7 @@ void gui_init(dt_iop_module_t *self)
   gtk_box_pack_start(GTK_BOX(self->widget), g->grey_point_source, TRUE, TRUE, 0);
   dt_bauhaus_slider_set_format(g->grey_point_source, "%.2f %%");
   gtk_widget_set_tooltip_text(g->grey_point_source, _("adjust to match the average luminance of the subject.\n"
-                                                      "except in back-lighting situations, this should be around 18%."));
+                                                      "except in back-lighting situations, this should be around 18 %%."));
```
Comment (TurboGit) [inline src/iop/filmic.c]: Sorry my bad, in tooltip there is no need to have double or quoted %. I'm about to commit a fix with this patchset. I have also removed spaces around \n.

**Comment by TurboGit** [comment]
Manually merged.

**Pull request closed without merging**